cli/lsp/config.rs

```diff
@@ -106,6 +106,101 @@ impl Default for CompletionSettings {
   }
 }
 
+#[derive(Debug, Default, Clone, Deserialize, Serialize, PartialEq, Eq)]
+#[serde(rename_all = "camelCase")]
+pub struct InlayHintsSettings {
```

Comment:
There is a vscode setting `"editor.inlayHints.enabled": "off"`. I'm guessing that would set all of these to false?

Reply:
@dsherret that setting determines if vscode requests inlay hints from extensions that support it.  It shouldn't factor into how we deal with it.  We broadcast we support them and we respond on `textDocument/inlayHints`.